File: server/client/src/main/kotlin/cz/uhk/mois/client/domain/Device.kt
package cz.uhk.mois.client.domain

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.Instant
import javax.persistence.*

@Table(name = "DEVICES")
class Device(
    @Id
    @GeneratedValue(strategy = GenerationType.TABLE)
    var id: Long? = null,

    @Column("typeId")
    var typeId: Long,

    @Column("modelName")
    var modelName: String,

    @Column("serialNumber")
    var serialNumber: String,

    @Column("releaseDate")
    var releaseDate: Instant?,
)

File: web/src/api/models/Device.ts
import { DeviceType, EMPTY_DEVICE_TYPE } from './DeviceType'

export type Device = {
    id: string
    typeId: string
    modelName: string
    serialNumber: string
    releaseDate: string
    type: DeviceType
}

export const EMPTY_DEVICE = {
    id: '',
    typeId: '',
    modelName: '',
    serialNumber: '',
    releaseDate: '',
    type: EMPTY_DEVICE_TYPE,
}
